spec/models/patient_spec.rb

```diff
@@ -206,5 +206,60 @@
       end
     end
   end
+
+  context '.discard_data' do
+    before do
+      create_list(:prescription_drug, 2, patient: patient)
+      create(:medical_history, patient: patient)
+      create_list(:appointment, 2, patient: patient)
+      bps = create_list(:blood_pressure, 2, patient: patient)
+      sugars = create_list(:blood_sugar, 2, patient: patient)
+      (bps + sugars).each do |record|
+        create(:encounter,
+               :with_observables,
+               patient: patient,
+               observable: record,
+               facility: patient.registration_facility)
+      end
+    end
+
+    it "should discard a patient's address" do
+      patient.discard_data
+      expect(patient.address.discarded?).to be true
```

Comment:
Nifty RSpec trick I learned recently - You can write this expectation as:

```ruby
expect(patient.address).to be_discarded
```

RSpec will take any `be_something` matcher and automatically call the corresponding query method on the object and expect the result to be true! I thought this was really cool when I came across it, just wanted to share 😄